playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -233,30 +249,59 @@ private MediaSource buildExoMediaSource(PKMediaSourceConfig sourceConfig) {
         if (mediaDataSourceFactory == null) {
             mediaDataSourceFactory = buildDataSourceFactory();
         }
+
+        if (subtitlesList != null) {
+            for (int subtitlePosition = 0 ; subtitlePosition < subtitlesList.size() ; subtitlePosition ++) {
+                mediaSources[subtitlePosition + 1] = buildSubtitleSource(subtitlesList.get(subtitlePosition));
+            }
+        }
+
         switch (format) {
 
             case dash:
                 if (manifestDataSourceFactory == null) {
                     manifestDataSourceFactory = buildDataSourceFactory();
                 }
-                return new DashMediaSource.Factory(
+                mediaSources[0] = new DashMediaSource.Factory(
                         new DefaultDashChunkSource.Factory(mediaDataSourceFactory),
                         manifestDataSourceFactory)
                         .createMediaSource(uri);
+                return new MergingMediaSource(mediaSources);
+
             case hls:
-                return new HlsMediaSource.Factory(mediaDataSourceFactory)
+                mediaSources[0] = new HlsMediaSource.Factory(mediaDataSourceFactory)
                         .createMediaSource(uri);
-            // mp4 and mp3 both use ExtractorMediaSource
+                return new MergingMediaSource(mediaSources);
+
+                // mp4 and mp3 both use ExtractorMediaSource
             case mp4:
             case mp3:
-                return new ExtractorMediaSource.Factory(mediaDataSourceFactory)
+                mediaSources[0] = new ExtractorMediaSource.Factory(mediaDataSourceFactory)
```

Comment:
please test it for mp4 ...

regarding mp3 not sure if we need to support it

Reply:
Yes tested with Demo entries json. It is working with mp4